plonk-core/src/proof_system/prover.rs

```diff
@@ -278,24 +268,29 @@ where
         w_r_poly = Self::add_blinder(&w_r_poly, n, 1);
         w_o_poly = Self::add_blinder(&w_o_poly, n, 1);
         w_4_poly = Self::add_blinder(&w_4_poly, n, 1);
+        let w_polys = [
+            label_polynomial!(w_l_poly),
+            label_polynomial!(w_r_poly),
+            label_polynomial!(w_o_poly),
+            label_polynomial!(w_4_poly),
+        ];
 
         // Commit to witness polynomials.
-        let w_l_poly_commit = KZG10::commit(commit_key, &w_l_poly, None, None)?;
-        let w_r_poly_commit = KZG10::commit(commit_key, &w_r_poly, None, None)?;
-        let w_o_poly_commit = KZG10::commit(commit_key, &w_o_poly, None, None)?;
-        let w_4_poly_commit = KZG10::commit(commit_key, &w_4_poly, None, None)?;
+        let (w_commits, w_rands) = PC::commit(commit_key, w_polys.iter(), None)
+            .map_err(to_pc_error::<F, PC>)?;
```

Comment:
shouldn't `.map_err(|e| Error::from(e))?;` do the work?

Reply:
It doesn't like impl `From` for associated types, I guess because it has no idea if there might exist another `From` impl for the actual type. I'm open for suggestions but I don't have a clean proposal other than what is implemented here.

Reply:
Will give it a thought. Filed #86